tests/cpptests/micro-benchmarks/benchmark_index_iterator.cpp

```diff
@@ -84,30 +84,30 @@ class BM_IndexIterator : public benchmark::Fixture {
     }
 
     void createIndex(IndexFlags flags) {
+        if (flags == (Index_DocIdsOnly | Index_Temporary)) {
+            // Special case reserved for `Index_DocIdsOnly` with raw doc IDs
+            RSGlobalConfig.invertedIndexRawDocidEncoding = true; // Enable raw doc ID encoding, until the benchmark's tearDown
+        }
+
         // Create a new InvertedIndex with the given flags
         size_t dummy;
         index = NewInvertedIndex(flags, 1, &dummy);
-        IndexEncoder encoder = InvertedIndex_GetEncoder(flags);
 
         if (flags == Index_StoreNumeric) {
             // Populate the index with numeric data
             for (size_t i = 0; i < ids.size(); ++i) {
                 InvertedIndex_WriteNumericEntry(index, ids[i], static_cast<double>(i));
             }
         } else if (flags == Index_DocIdsOnly) {
-            // Populate the index with document IDs only
-            for (size_t i = 0; i < ids.size(); ++i) {
-                RSIndexResult rec = {.docId = ids[i], .data = {.tag = RSResultData_Virtual}};
-                InvertedIndex_WriteEntryGeneric(index, encoder, &rec);
+            if (flags == (Index_DocIdsOnly | Index_Temporary)) {
+                // Ensure we are using the raw doc ID encoder
+                RS_ASSERT_ALWAYS(InvertedIndex_GetEncoder(InvertedIndex_Flags(index)) != InvertedIndex_GetEncoder(Index_DocIdsOnly));
```

Comment:
This assertion is unreachable (if we are here, `flags` is never `(Index_DocIdsOnly | Index_Temporary)`).
Let's move it to the beginning of `createIndex`